Sources/NIOPosix/Bootstrap.swift

```diff
@@ -326,6 +326,28 @@ public final class ServerBootstrap {
         return self.bind(unixDomainSocketPath: unixDomainSocketPath)
     }
 
+    #if canImport(Darwin) || os(Linux)
+    /// Bind the `ServerSocketChannel` to a VSOCK socket.
+    ///
+    /// - parameters:
+    ///   - vsockAddress: The VSOCK socket address to bind on.
+    public func bind(to vsockAddress: VsockAddress) -> EventLoopFuture<Channel> {
```

Comment:
this is public API so shouldn't be `#if os(...)`'d. In the body you can `#if os(...)` and just return an unsupported operation error.

Reply:
> withBoundSocket(descriptor:) which does use #if !os(...) guard to not include it in the API.

`withBoundSocket` is an interesting case because it's an overload. There is always

```
public func withBoundSocket(_ socket: NIOBSDSocket.Handle) -> EventLoopFuture<Channel>
```

and on non-Windows there is also the deprecated

```
        @available(*, deprecated, renamed: "withBoundSocket(_:)")
        public func withBoundSocket(descriptor: CInt) -> EventLoopFuture<Channel> {
```

When Windows support was started, we wanted to migrate everybody to the new, former one because it works everywhere but because of public API we could only deprecate it of course (and not remove it). But because Windows had not been supported at all, we could actually remove it (by `#if`) on Windows.

Was it 100% necessary to `#if` it out on Windows? No. But it's also not a big issue because the API still exists, just with slightly different types. So I think that makes some sense.